code/campaignclassic/src/main/res/layout/push_template_filmstrip_carousel.xml

```diff
@@ -15,45 +18,59 @@
         android:id="@+id/notification_title"
         style="@style/Title"
         android:layout_width="match_parent"
-        android:layout_height="20dp"
+        android:layout_height="wrap_content"
         android:layout_gravity="start"
         android:textAlignment="viewStart"
+        app:autoSizeTextType="uniform"
+        app:autoSizeMinTextSize="6sp"
+        app:autoSizeMaxTextSize="20sp"
+        app:autoSizeStepGranularity="2sp"
         android:layout_toLeftOf="@+id/large_icon"/>
 
     <TextView
         android:id="@+id/notification_body_expanded"
         style="@style/Body"
         android:layout_width="match_parent"
-        android:layout_height="30dp"
+        android:layout_height="wrap_content"
         android:layout_below="@id/notification_title"
         android:layout_gravity="start"
         android:textAlignment="viewStart"
+        app:autoSizeTextType="uniform"
+        app:autoSizeMinTextSize="6sp"
+        app:autoSizeMaxTextSize="20sp"
+        app:autoSizeStepGranularity="2sp"
         android:layout_toLeftOf="@+id/large_icon"/>
 
     <ImageView
         android:id="@+id/manual_carousel_filmstrip_center"
-        android:scaleType="centerCrop"
+        android:scaleType="fitCenter"
         android:adjustViewBounds="true"
         android:layout_width="200dp"
         android:layout_height="200dp"
         android:layout_gravity="center"
-        android:layout_centerInParent="true"
+        android:layout_centerHorizontal="true"
         android:layout_below="@id/notification_body_expanded">
     </ImageView>
 
     <TextView
         android:id="@+id/manual_carousel_filmstrip_caption"
-        style="@style/TextAppearance.Compat.Notification.Title"
+        style="@style/Caption"
         android:layout_width="wrap_content"
-        android:layout_height="20dp"
+        android:layout_height="wrap_content"
         android:layout_below="@id/manual_carousel_filmstrip_center"
         android:layout_centerHorizontal="true"
-        android:layout_gravity="center" />
+        android:layout_gravity="center"
+        app:autoSizeTextType="uniform"
+        app:autoSizeMinTextSize="6sp"
+        app:autoSizeMaxTextSize="20sp"
+        app:autoSizeStepGranularity="2sp"/>
 
     <ImageView
         android:id="@+id/manual_carousel_filmstrip_left"
         android:scaleType="centerCrop"
         android:adjustViewBounds="true"
+        android:layout_gravity="center"
+        android:layout_centerHorizontal="true"
```

Comment:
Should this be `layout_centerVertical=true`? Also is `layout_gravity=center` needed?

Reply:
`android:layout_centerHorizontal="true"` is required for the center carousel image to correctly align in the middle of the template remote view. `layout_gravity=center` is not needed and has been removed.